src/plugins/data_source/server/plugin.ts

```diff
@@ -107,4 +150,28 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
       };
     };
   };
+
+  private async logAuditMessage(
+    auditorPromise: Promise<Auditor>,
+    dataSourceId: string,
+    request: OpenSearchDashboardsRequest
+  ) {
+    const auditor = await auditorPromise;
+    const auditMessage = this.getAuditMessage(request, dataSourceId);
+
+    auditor.add({
+      message: auditMessage,
+      type: 'opensearch.dataSourceClient.call.internalUser',
+    });
+  }
+
+  private getAuditMessage(request: OpenSearchDashboardsRequest, dataSourceId: string) {
+    const rawRequest = ensureRawRequest(request);
+    const remoteAddress = rawRequest?.info?.remoteAddress;
+    const xForwardFor = request.headers['x-forwarded-for'];
```

Comment:
why is remoteAddress and `x-forwarded-for` needed ?

Reply:
to log the requester's info